### DISCUSSION TITLE
Scalability compared to Websockets?

### DISCUSSION THREAD

**Discussion by ggg7125** [geckosio/geckos.io#186] [General]
Which has more scalability? I'm particularly looking at Geckos vs uWebsockets. Thank you. (I would like to have 100 players running around together at once)

**Comment by jonit-dev** [comment]
I guess that this depends on your code architecture, mostly.

For example, I'm developing an online RPG using Geckos.io and I'm dividing the area where our data packages are sent by player cameraCoordinates zones.

In other words, if a character moves in a particular X,Y position, the server receives the package and then it decides to which player it's going to redirect this data to (only the ones inside the emitter camera!)

**Comment by ggg7125** [comment]
I found this stress test from 1.5 years ago: https://github.com/geckosio/stress-test
The amount of players it can handle does not seem to be very high. Only 44 connections @ 30 fps.
I guess WebRTC is good for shooters or other games with low concurrent players, but I think this means I should choose websockets if I'm trying to do a 100+ player RPG. Or am I wrong?

**Reply by jonit-dev** [reply]
It's currently using node-datachannel, which has better performance than WebRTC.
```Version 2 has huge performance improvements. I switched from [wrtc](https://www.npmjs.com/package/wrtc) to [node-datachannel](https://www.npmjs.com/package/node-datachannel), which is much lighter and faster compared to wrtc.```

btw, I'm doing an MMORPG engine in Typescript/Phaser/Node/React.

I'll leave an invite here if you want to participate: https://join.slack.com/t/rpg-online-nft/shared_invite/zt-14jrc35ia-SkKQbukmJgUxFsrduK4m2A

There are more devs involved

![image](https://user-images.githubusercontent.com/30300459/158211458-2c567585-729b-4312-9c63-c1d79137095b.png)

**Reply by yandeu** [reply]
> Only 44 connections @ 30 fps

I just tested the latest version (2.1.8) with the stress-test and got **128 connections @ 60 fps**.

**Reply by jonit-dev** [reply]
@yandeu do you know what's the maximum thereshold? 
Can it hold like, 1000 players online?

Any thoughts on implementing [cluster mode like Socket.IO](https://socket.io/docs/v4/pm2/)? I guess it would significantly help our performance.

Thanks for the awesome lib!
